sql/src/main/java/io/crate/operation/tablefunctions/GenerateSeriesFunction.java

```diff
@@ -0,0 +1,279 @@
+
+    private static final Map<DataType, FunctionImplementation> implTwoArgs;
+    private static final Map<DataType, FunctionImplementation> implThreeArgs;
+
+    static {
+        implTwoArgs = new HashMap<>();
+        implThreeArgs = new HashMap<>();
+
+        DataType type = DataTypes.INTEGER;
+
+        implTwoArgs.put(type, new IntegerGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type)), type, FunctionInfo.Type.TABLE)));
+        implThreeArgs.put(type, new IntegerGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type, type)), type, FunctionInfo.Type.TABLE)));
+
+        type = DataTypes.LONG;
+
+        implTwoArgs.put(type, new LongGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type)), type, FunctionInfo.Type.TABLE)));
+        implThreeArgs.put(type, new LongGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type, type)), type, FunctionInfo.Type.TABLE)));
+
+        type = DataTypes.FLOAT;
+
+        implTwoArgs.put(type, new FloatGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type)), type, FunctionInfo.Type.TABLE)));
+        implThreeArgs.put(type, new FloatGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type, type)), type, FunctionInfo.Type.TABLE)));
+
+        type = DataTypes.DOUBLE;
+
+        implTwoArgs.put(type, new DoubleGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type)), type, FunctionInfo.Type.TABLE)));
+        implThreeArgs.put(type, new DoubleGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type, type)), type, FunctionInfo.Type.TABLE)));
+    }
+
+
+    public static void register(TableFunctionModule module){
+        module.register(NAME, new BaseFunctionResolver(
+            Signature.numArgs(2,3).and(Signature.withLenientVarArgs(Signature.ArgMatcher.NUMERIC))) {
+
+            @Override
+            public FunctionImplementation getForTypes(List<DataType> dataTypes) throws IllegalArgumentException {
+                Map<DataType, FunctionImplementation> impl = dataTypes.size() == 2 ? implTwoArgs : implThreeArgs;
+                DataType first = dataTypes.get(0);
+                DataType third = dataTypes.size() > 2 ? dataTypes.get(2) : DataTypes.INTEGER;
+
+                DataType implType;
+                if(NUMERIC_WITH_DECIMAL.contains(first) || NUMERIC_WITH_DECIMAL.contains(third))
+                    implType = dataTypes.contains(DataTypes.DOUBLE) || dataTypes.contains(DataTypes.LONG) ? DataTypes.DOUBLE : DataTypes.FLOAT;
+                else
+                    implType = dataTypes.contains(DataTypes.LONG) ? DataTypes.LONG : DataTypes.INTEGER;
```

Comment:
don't you need to check for double here too, since the second can be a double, right?